icu4c/source/test/intltest/uniquecharstrtst.cpp

```diff
@@ -0,0 +1,53 @@
+// © 2023 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <memory>
+
+#include "cmemory.h"
+#include "cstring.h"
+#include "uniquecharstr.h"
+#include "uniquecharstrtst.h"
+
+UniqueCharStringsTest::UniqueCharStringsTest()
+{
+}
+
+UniqueCharStringsTest::~UniqueCharStringsTest()
+{
+}
+
+void UniqueCharStringsTest::runIndexedTest( int32_t index, UBool exec, const char* &name, char* /*par*/ )
+{
+    TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(TestBasic);
+    TESTCASE_AUTO_END;
+}
+
+void UniqueCharStringsTest::TestBasic() {
+    UErrorCode status = U_ZERO_ERROR;
+    UniqueCharStrings strings(status);
+    if (U_FAILURE(status)) {
+       errln("failed to construct UniqueCharStrings - %s", u_errorName(status));
+    }
+    UnicodeString key1("abcde", -1, US_INV);
+    UnicodeString key2(u"fgihjk", 2); // the key should be only "fg" since the
+                                      // length is 2.
+    UnicodeString key3("lmn", -1, US_INV);
+    int32_t index1 = strings.add(key1, status);
+    if (U_FAILURE(status)) {
```

Comment:
Come to think of it, you could also use `assertSuccess()` here to make things a little simpler...